### PR TITLE
system fields: relations

### PR DESCRIPTION
**Pull request by jrcastro2**
* Adds vocabulary field to be able to
  dereference one value from an object
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/695

### CODE REVIEW

File: invenio_records/systemfields/relations/relations.py [hunk 1]
```diff
@@ -125,34 +126,65 @@ class ListRelation(RelationBase):
...
+        :type relation_field: str
+        """
+        self.relation_field = relation_field
+        super().__init__(*args, **kwargs)
+
     def parse_value(self, value):
         """Parse a record (or ID) to the ID to be stored."""
         if isinstance(value, (tuple, list)):
+            if self.relation_field:
+                return [super(ListRelation, self).parse_value(
```
Comment (lnielsen) [inline invenio_records/systemfields/relations/relations.py:145]:
```suggestion
                return [super().parse_value(
```

File: invenio_records/systemfields/relations/relations.py [hunk 2]
```diff
@@ -125,34 +126,65 @@ class ListRelation(RelationBase):
...
+                    v.get(self.relation_field)) for v in
+                        value if self.relation_field in v]
             return [super(ListRelation, self).parse_value(v) for v in value]
```
Comment (lnielsen) [inline invenio_records/systemfields/relations/relations.py:148]:
```suggestion
            return [super().parse_value(v) for v in value]
```

File: invenio_records/systemfields/relations/relations.py [hunk 3]
```diff
@@ -125,34 +126,65 @@ class ListRelation(RelationBase):
...
         else:
             raise InvalidRelationValue('Invalid value. Expected list.')
 
+    def _get_parent(self, record, keys):
+        """Get parent record."""
```
Comment (lnielsen) [inline invenio_records/systemfields/relations/relations.py:153]:
```suggestion
        """Get parent dict."""
```

File: invenio_records/systemfields/relations/relations.py [hunk 4]
```diff
@@ -125,34 +126,65 @@ class ListRelation(RelationBase):
...
+                        )
+                parent = parent[k]
+        return parent
+
     def set_value(self, record, value):
         """Set the relation value."""
         store_values = self.parse_value(value)
         # Validate all values
-        if self.exists_many(store_values):
+        if self.exists_many(store_values) and self.relation_field:
```
Comment (lnielsen) [inline invenio_records/systemfields/relations/relations.py:174]: I would rewrite this function a bit.
- ``language_field`` i guess is a left-over
- ensure to have as little code as possible that's different between the two versions.


```python
def set_value(self, record, value):
        """Set the relation value."""
        store_values = self.parse_value(value)
        
        # Validate all values
        if self.exists_many(store_values):
            keys = parse_lookup_key(self.value_key)
            store_key, rel_id_key = keys[-2:]
            parent = self._get_parent(record, keys)

            # Create the list of values to store.
            if self.relation_field:
                values_list = deepcopy(value)
                for v in values_list:
                    sv = v.get(self.relation_field)
                    if sv:
                        v[self.relation_field] = {rel_id_key: sv}
            else:
                values_list = [{rel_id_key: sv} for sv in store_values]
                
            parent[store_key] = values_list
        else:
            raise InvalidRelationValue("Invalid values.")
```

**Review by lnielsen — APPROVED** [review]
I left some minor comments. Overall everything looks good. Well-done!

File: invenio_records/systemfields/relations/relations.py (hunk 1 above)
Comment (jrcastro2) [reply invenio_records/systemfields/relations/relations.py:145]: Can't do this change or we get this error `TypeError: super(type, obj): obj must be an instance or subtype of type`.

File: invenio_records/systemfields/relations/relations.py (hunk 2 above)
Comment (jrcastro2) [reply invenio_records/systemfields/relations/relations.py:148]: Same as above.

**Pull request merged**